app/templates/play/campaign-view.jade

```diff
@@ -220,26 +223,32 @@ if view.showAds()
 
   .game-controls.header-font.picoctf-hide
     if !(features.picoCtf || features.brainPop)
-      button.btn.poll.hidden(data-i18n="[title]play.poll")
-      if !features.codePlay
+      if view.shouldShow('poll')
+        button.btn.poll.hidden(data-i18n="[title]play.poll")
+      if view.shouldShow('clans')
         a.btn.clans(href="/clans", data-i18n="[title]clans.clans")
-      button.btn.items(data-toggle='coco-modal', data-target='play/modal/PlayItemsModal', data-i18n="[title]play.items")
-      button.btn.heroes(data-toggle='coco-modal', data-target='play/modal/PlayHeroesModal', data-i18n="[title]play.heroes")
-      button.btn.achievements(data-toggle='coco-modal', data-target='play/modal/PlayAchievementsModal', data-i18n="[title]play.achievements")
-      if (me.get('anonymous') === false || me.get('iosIdentifierForVendor') || isIPadApp) && !me.freeOnly()
+      if view.shouldShow('items')
+        button.btn.items(data-toggle='coco-modal', data-target='play/modal/PlayItemsModal', data-i18n="[title]play.items")
+      if view.shouldShow('heros')    
+        button.btn.heroes(data-toggle='coco-modal', data-target='play/modal/PlayHeroesModal', data-i18n="[title]play.heroes")
+      if view.shouldShow('achievements')
+        button.btn.achievements(data-toggle='coco-modal', data-target='play/modal/PlayAchievementsModal', data-i18n="[title]play.achievements")
+      if view.shouldShow('buy-gems')
         button.btn.gems(data-toggle='coco-modal', data-target='play/modal/BuyGemsModal', data-i18n="[title]play.buy_gems")
-      if !me.get('anonymous', true) && !features.codePlay
+      if !me.get('anonymous', true) && view.shouldShow('settings')
```

Comment:
Make sure this works as expected with a few feature modes. Good to refactor, but want to make sure third parties see no difference in behavior.